src/NLog/Config/ConfigurationItemFactory.cs

```diff
@@ -269,9 +269,9 @@ private static ConfigurationItemFactory BuildDefaultFactory()
                     factory.RegisterItemsFromAssembly(extensionAssembly);
                     success = true;
                 }
-                catch (Exception)
+                catch (Exception ex)
                 {
-                    InternalLogger.Warn("Auto loading assembly file: {0} failed! Skipping this file.", extensionDll);
+                    InternalLogger.Warn(ex, "Auto loading assembly file: {0} failed! Skipping this file.", extensionDll);
```

Comment:
backwardscompatible is really important, so I keep this as is, but I will add an note :)